proxy/src/registry.rs

```diff
@@ -11,7 +11,7 @@ use std::str::FromStr;
 use radicle_registry_client::{self as protocol, ClientT, CryptoPair};
 pub use radicle_registry_client::{
     parse_ss58_address, AccountId, Balance, BlockHash, Id, IdStatus, ProjectDomain, ProjectName,
-    MINIMUM_FEE,
+    MINIMUM_TX_FEE, REGISTRATION_FEE,
```

Comment:
Maybe `HANDLE_REGISTRATION_FEE` would be more descriptive, as we also call the transaction a "registration" for projects and members.

Reply:
I feel that. `HANDLE_REGISTRATION_FEE` is not appropriate since it's not common language in the registry neither for orgs. Since we call users and orgs "domains" we could call it `DOMAIN_REGISTRATION_FEE`? 

Having this said I would prefer to keep it called this way, shall we broaden the scope of this fee. I am fine with the notion that the registration fee only applies to user and org registration.

What do you think?

Reply:
I am willing to accept the name, but what might actually be the problem here is that in the current implementation, the ui "decides" when to show the registration fee. I would think this is a registry or proxy concern. SO maybe we could actually keep the way `deposits` were split up by transaction type and this way pass to the ui which transaction types add a registration fee. This preserves being flexible for future adjustments.

Reply:
> I am willing to accept the name, but what might actually be the problem here is that in the current implementation, the ui "decides" when to show the registration fee. I would think this is a registry or proxy concern. SO maybe we could actually keep the way `deposits` were split up by transaction type and this way pass to the ui which transaction types add a registration fee. This preserves being flexible for future adjustments.

This is a very good observation. I will give it some thought, I think I have an idea already.

Reply:
Let's quickly give a step back.

### Considerations

1. How do we adjust to changes to the scope of the registration fee?
    say, we want project registration to incur in the registration fee
2. How do we adjust in case we need a specific registration fee value per type of transaction?

We should keep these in mind when evaluating the current solution and different suggestions, though we do not see any changes in this field coming up.

### Now (Option 0)

- The Registry exposes `REGISTRATION_FEE`, a fee that is withdrawn when registering users and org (not projects or adding members to an org).
- Proxy provides `registrationFee` to the `ui` through `session`
- each registration screen (`UserRegistration`, `OrgRegistration`, `MemberAddition`, `ProjectRegistration` passes this universal `registrationFee` value to the `Transaction` component
- `Transaction` simply delegates the received universal `registrationFee` value to the `costSummary` function
- `costSummary` looks at the received transaction/message and decides to include the `registrationFee` in the returned `CostSummary` or not based on the message.
- `Transaction` shows or hides the `registration-fee` `Row` based on `registrationFee` being defined in the `summary`

**Regarding the considerations**

1. We would only need to change the `paysRegistrationFee`
2. We would need to change what we pass from the registration screens down to the `Transaction` component, possibly a struct with the breakdown of the fee value per tx type, as we had for the deposits. Then, in `costSummary`, we would look up the appropriate value. 
    - This is not something we see coming at all, to keep in mind.

**Observations**:

- Kind of confusing that we are passing the `registrationFee` value from the registration screens down to the `Transaction` component even for screens that don't incur into that fee.

### Option A

- (same) Registry exposes `REGISTRATION_FEE`, a fee that is withdrawn when registering users and org (not projects or adding members to an org).
- (same) Proxy provides `registrationFee` to the `ui` through `session`
- (**change**) each registration screen (`UserRegistration`, `OrgRegistration`, `MemberAddition`, `ProjectRegistration`  pass `session` to the `Transaction` component. `Transaction` then looks up the session's `registrationFee` value, passing it to `costSummary`
- (same)`costSummary` looks at the received transaction/message and decides to include the `registrationFee` in the returned `CostSummary` or not based on the message.
- (same) `Transaction` shows or hides the `registration-fee` `Row` based on `registrationFee` being defined in the `summary`

**Regarding the considerations**

1. We would only need to change the `paysRegistrationFee`
2. Same as for `Option 0`

**What does this improve?**

- Drops the confusion reported in `Option 0` where registration screens pass the `registrationFee` value even when unsuited.

### Option B

- (same) Registry exposes `REGISTRATION_FEE`, a fee that is withdrawn when registering users and org (not projects or adding members to an org).
- (**change**) Proxy provides `registrationFee` **struct** to the `ui` through `session`:

    ```rust
    struct RegistrationFee {
      user: Option<Balance>,
      org: Option<Balance>,
      memberAddition: Option<Balance>,
      project: Option<Balance>,
    }
    ```

As of now, it would be built as:

```rust
RegistrationFee {
  user: Some(registry::REGISTRATION_FEE),
  org: Some(registry::REGISTRATION_FEE),
  memberAddition: None,
  project: None,
}
```

- (**change**) each registration screen (`UserRegistration`, `OrgRegistration`, `MemberAddition`, `ProjectRegistration` passes their respective entry in the above `RegistrationFee` struct to the `Transaction` component, which represents whether that registration incurs in a registration fee or not and how much it is if yes.
- (**change**) `costSummary` looks at the received transaction/message and decides to include the `registrationFee` in the returned `CostSummary` based on the `registrationFee?` param
- (same) `Transaction` shows or hides the `registration-fee` `Row` based on `registrationFee` being defined in the `summary`

**Regarding the considerations**

1. We would only need to change the appropriate `RegistrationFee` instance field values in the proxy
2. We would need to expose more granular values from the Registry for each tx type and instantiate the `RegistrationFee` struct accordingly in the proxy. The UI is left untouched. 

**What does this improve?**

- It reduces the confusion around a registration screen passing the universal `registrationFee` value to `Transaction` even when the said screen does not incur in the registration fee.
- Injects data dependencies top-down
- Super flexible

- **Issues**
    - With this setup, we offer more flexibility than needed, which can be confusing and anti-YAGNI
    - making the view over which screens incur in the registration fee more implicit and spread (vs having a `paysRegistrationFee` that unifies it).
    - A screen can pass the wrong registration fee value down to `Transaction` (i.e, while having it unified in `paysRegistrationFee` ensures one source of truth)

### Conclusion

All considered I think that `Option A` would solve the confusion around screens that don't incur in the registration fee passing _the_ registration fee value down to `Transaction` without creating unnecessary machinery.

I believe `Option B` is what you (@MeBrei) were suggesting. I am ok with moving there, though I don't think it's a straightforward trade-off as it introduces more code that does not serve any current or foreseeable use case.

Let me know what you think :)

Reply:
Thank you for the thorough comparison!

I would still prefer `Option B` as I think the ui should not have to have "insider" knowledge how transactions are handled in the registry. More specifically, if we ever want to show "historic" transactions, the registration fee should actually be part of the transaction data, as it might change over time.

BUT as a first iteration I think the current implementation could be merged to keep things moving